pages/db/[language]/index.tsx

```diff
@@ -17,6 +21,14 @@ export { getSourceLanguageStaticPaths as getStaticPaths } from "utils/nextJsHelp
 
 export default function DbIndexPage() {
   const { sourceLanguageName, sourceLanguage } = useDbQueryParams();
+  const availableViews = useAvailableDbViews();
+  const [currentView, setCurrentView] = useAtom(currentViewAtom);
+
+  React.useEffect(() => {
```

Comment:
what does this effect do? Can you explain why this being done in the index page?

Reply:
AFAICT there's only one scenario when the unavailable view issue could arise:

1. user navigates to a pli or chn text in numbers view, _then_
2. user switches DB to skt or tib (ie. navigates to page generated by `pages/db/[language]/index.tsx`)
3. user selects a text but encounters an error/blank resuts because numbers view doesn't exists

Making sure the view is set to the default on the DB landing page ensures an existing view will always be served once a text is selected.

Further, there is no other context in which resetting the view makes sense (if the user is changing text "within" a DB, available view options are already appropriate for the given DB)

Reply:
Further to the above, as I understand the received wisdom is that `useEffect` is evil, I moved the check to the DB selector handler to remove the `useEffect` in https://github.com/BuddhaNexus/buddhanexus-frontend-next/pull/178/commits/bf32075e50ebd61fb5c459d49d1253cb934dcdb6.

Reply:
I've never heard that useEffect is evil! But I suppose it introduces async logic, which is a bit more tricky.

Also, I guess in nextjs we just need to know that all `useEffect` hooks run on the client side, which means this can't be prerendered. At least in the pages router, not sure about the app router :)

Reply:
I've never heard that useEffect is evil! But I suppose it introduces async logic, which is a bit more tricky.

Also, I guess in nextjs we just need to know that all useEffect hooks run on the client side, which means this can't be prerendered. At least in the pages router, not sure about the app router :)

Reply:
Good considerations, thanks!

My YouTube dude gave the "is evil" line; the more measured version would be, if you don't have do, don't.